src/libopensc/pkcs15-pin.c

```diff
@@ -367,6 +367,7 @@ int sc_pkcs15_verify_pin(struct sc_pkcs15_card *p15card,
 	r = sc_pin_cmd(card, &data, &auth_info->tries_left);
 	sc_log(ctx, "PIN cmd result %i", r);
 	if (r == SC_SUCCESS)
+	if (r == SC_SUCCESS && pin_obj->usage_counter == 0)
```

Comment:
Look at opensc.conf and start with comment "Use PIN caching?"
    
    On 6/26/2016 7:35 AM, Nuno Goncalves
      wrote:
    
    
      In src/libopensc/pkcs15-pin.c:
      > @@ -367,6 +367,7 @@ int sc_pkcs15_verify_pin(struct sc_pkcs15_card *p15card,
>  	r = sc_pin_cmd(card, &data, &auth_info->tries_left);
>  	sc_log(ctx, "PIN cmd result %i", r);
>  	if (r == SC_SUCCESS)
> +	if (r == SC_SUCCESS && pin_obj->usage_counter == 0)

      Since I'm testing this with pkcs11-tool, all is OpenSC code. I
        couldn't find where the tool or pkcs11 layer is caching the pin.
        Can you point me out where is it coming from so we can fix it?
      —
        You are receiving this because you are subscribed to this
        thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
    
    
    -- 

 Douglas E. Engert  <DEEngert@gmail.com>